CryptoPrices/Sources/Styleguide/Sources/Styleguide/ButtonStyles/SecondaryButtonStyle.swift

```diff
@@ -0,0 +1,28 @@
+//
+//  SecondaryButtonStyle.swift
+//  Styleguide
+//
+//  Created by Doan Thieu on 10/01/2023.
+//
+
+import SwiftUI
+
+public struct SecondaryButtonStyle: ButtonStyle {
+
+    private static let backgroundColor = Colors.carnation.swiftUIColor
+    private static let selectedBackgroundColor = Colors.carnation.swiftUIColor.opacity(0.5)
```

Comment:
Do we have any other purpose for using `static let`, except for defining it as a constant? Maybe only `let` is enough? Thank you 🙏 .

Reply:
@nkhanh44  That's a good question 🙏

The `static let` make these properties _type property_, not _instance property_. So regardless of how many instances of `SecondaryButtonStyle` we create, these properties are created once, and we'll save a little memory here since the compiler won't have to allocate memories on the stack for these for every instance.

This is usually used when we want to share data between instances, especially when the data is expensive to create. In Swift, `static` also makes data lazily initialized. Let's say we want to add a `DateFormatter` (not make sense here, just an example):
```swift
public struct SecondaryButtonStyle: ButtonStyle {

  private static var dateFormatter: DateFormatter = {
    let formatter = DateFormatter()
    formatter.dateStyle = .medium
    // ... other config
    return formatter
  }()
}
```

In any _instance method_, we can access through `SecondaryButtonStyle.dateFormatter` or the preferred way is `Self. dateFormatter` (with the uppercased `S`). (In _static method_, just use `dateFormatter` directly). Hope this clears your concern 🙏

Reply:
Thank you  🙏. It's clear to me.